src/transformation/visitors/class/index.ts

```diff
@@ -86,10 +85,6 @@ function transformClassLikeDeclaration(
     const superInfo = getOrUpdate(classSuperInfos, context, () => []);
     superInfo.push({ className, extendedTypeNode });
 
-    if (extendedType) {
```

Comment:
Are you sure this can be removed?  Why does this not break promises construction?

Reply:
The check for lualib type has been moved to transformIdentifier. For a class extends, transformIdentifier is eventually called in createClassSetup.